doc-gen/src/classes/SidebarBuilder.ts

```diff
@@ -0,0 +1,367 @@
+      );
+
+      this.entries.push({
+        name: methodName,
+        path: relativePath,
+        type: "class",
+        className: importedName,
+      });
+    } else {
+      // Handle functions/hooks/components - same logic as generateFunctionDocs
+      const functionName = getFunctionName(node, importedName);
+
+      let entryType: "function" | "hook" | "component";
+      let typeDir: "functions" | "hooks" | "components";
+
+      if (importedName.startsWith("use")) {
+        entryType = "hook";
+        typeDir = "hooks";
+      } else if (isTsx) {
+        entryType = "component";
+        typeDir = "components";
+      } else {
+        entryType = "function";
+        typeDir = "functions";
+      }
+
+      const relativePath = this.generateDocsPath(
+        outputFilePath,
+        typeDir,
+        functionName,
+      );
+
+      this.entries.push({
+        name: functionName,
+        path: relativePath,
+        type: entryType,
+      });
+    }
+  }
+
+  private generateDocsPath(
+    outputFilePath: string,
+    typeDir: "functions" | "hooks" | "components" | "",
+    fileName: string,
+  ): string {
+    const fullPath = typeDir
+      ? `${outputFilePath}/${typeDir}/${fileName}.mdx`
+      : `${outputFilePath}/${fileName}.mdx`;
+
+    // Convert the output filepath to the final filepath by removing absolute path up to & including `docs/` and replacing with `wallets/`
+    const docsMatch = fullPath.match(/docs\/(.+)$/);
+    if (!docsMatch) {
+      throw new Error(`Could not find docs match for ${fullPath}`);
+    }
+
+    return `wallets/${docsMatch[1]}`;
+  }
+
+  private groupClassEntries(): Record<string, SidebarEntry[]> {
+    const classEntries: Record<string, SidebarEntry[]> = {};
+
+    for (const entry of this.entries.filter((e) => e.type === "class")) {
+      if (!entry.className) continue;
+
+      if (!classEntries[entry.className]) {
+        classEntries[entry.className] = [];
+      }
+      classEntries[entry.className].push(entry);
+    }
+
+    return classEntries;
+  }
+
+  private generateReactSdkSection(): string {
+    const componentEntries = this.entries.filter((e) => e.type === "component");
+    const hookEntries = this.entries.filter((e) => e.type === "hook");
+    const functionEntries = this.entries.filter((e) => e.type === "function");
+    const classEntries = this.groupClassEntries();
+
+    let yaml = `section: ${sdkReferenceSection}\n`;
+    yaml += `path: wallets/pages/reference/${this.packageName}/index.mdx\n`;
+    yaml += `contents:\n`;
+
+    if (componentEntries.length > 0) {
+      yaml += `  - section: Components\n`;
+      yaml += `    contents:\n`;
+
+      componentEntries.sort((a, b) => a.name.localeCompare(b.name));
+      for (const entry of componentEntries) {
+        yaml += `      - page: ${entry.name}\n`;
+        yaml += `        path: ${entry.path}\n`;
+      }
+    }
+
+    for (const [groupName, hookNames] of Object.entries(hookGroupings)) {
```

Comment:
we should also handle any hooks that don't belong to groupings and have them added to a `Other Hooks` group for now

Reply:
Good catch, there actually are a few